enterprise/kernel/src/main/java/org/neo4j/kernel/enterprise/builtinprocs/EnterpriseBuiltInDbmsProcedures.java

```diff
@@ -527,4 +524,14 @@ public QueryTerminationResult( QueryId queryId, String username )
             this.connectionCount = connectionCount;
         }
     }
+
+    public static class MetadataResult
+    {
+        public final Map<String,Object> metadata;
```

Comment:
The Neo4j results objects are kind-of a streamed version of a map, so work better for memory and for display in tools that already work with results, like the browser, neo4j-shell and cypher-shell. Only slightly less convenient for a client developer just wanting to pull the entire map and do a single lookup, which is what you are optimizing for.